src/cosmospy/_transaction.py

```diff
@@ -21,87 +26,149 @@ class Transaction:
         signature_compact = privkey.sign_deterministic(
-            message_bytes, hashfunc=hashlib.sha256, sigencode=ecdsa.util.sigencode_string_canonize
+            self._get_sign_doc().SerializeToString(),
+            hashfunc=hashlib.sha256,
+            sigencode=ecdsa.util.sigencode_string_canonize,
         )
+        return signature_compact
+
+    def _get_sign_doc(self):
+        sign_doc = tx.SignDoc()
+        sign_doc.body_bytes = self._tx_body.SerializeToString()
+        sign_doc.auth_info_bytes = self._get_auth_info().SerializeToString()
+        sign_doc.chain_id = self._chain_id
+        sign_doc.account_number = self._account_num
+        return sign_doc
 
-        signature_base64_str = base64.b64encode(signature_compact).decode("utf-8")
-        return signature_base64_str
-
-    def _get_sign_message(self) -> dict[str, Any]:
-        return {
-            "chain_id": self._chain_id,
-            "account_number": str(self._account_num),
-            "fee": {
-                "gas": str(self._gas),
-                "amount": [{"amount": str(self._fee), "denom": self._fee_denom}],
-            },
-            "memo": self._memo,
-            "sequence": str(self._sequence),
-            "msgs": self._msgs,
+    def _get_auth_info(self):
+        _auth_info = tx.AuthInfo()
+        _auth_info.signer_infos.append(self._get_signer_infos(self._get_pubkey()))
+        _auth_info.fee.gas_limit = self._gas
+        if self._fee_account:
+            _auth_info.fee.granter = self._fee_account
+        _auth_info.fee.amount.append(self._get_fee())
+        return _auth_info
+
+    def _get_fee(self):
+        _fee = coin.Coin()
+        _fee.amount = str(self._fee)
+        _fee.denom = self._fee_denom
+        return _fee
+
+    def _get_pubkey(self):
+        pubkey_bytes = privkey_to_pubkey(self._privkey)
+        _pubkey = pubkey.PubKey()
+        _pubkey.key = pubkey_bytes
+        return _pubkey
+
+    def _get_signer_infos(self, _pubkey):
+        signer_infos = tx.SignerInfo()
+        signer_infos.sequence = self._sequence
+        signer_infos.public_key.Pack(_pubkey)
+        signer_infos.public_key.type_url = "/cosmos.crypto.secp256k1.PubKey"
+        signer_infos.mode_info.single.mode = 1
+        return signer_infos
+
+    def get_api_pushable(self):
+        api_sync_modes = {
+            "broadcast_tx_sync": "BROADCAST_MODE_SYNC",
+            "broadcast_tx_async": "BROADCAST_MODE_ASYNC",
+            "broadcast_tx_commit": "BROADCAST_MODE_BLOCK"
         }
+        return json.dumps(
+            {"tx_bytes": self._get_tx_bytes(), "mode": api_sync_modes[self._sync_mode]}
+        )
+
+    def get_rpc_pushable(self):
+
+        return json.dumps(
+            {
+                "jsonrpc": "2.0",
+                "id": 1,
+                "method": self._sync_mode,
+                "params": {
+                    "tx": self._get_tx_bytes()
+                }
+            }
+        )
```

Comment:
```suggestion
```

Let's do as I suggested [here](https://github.com/hukkin/cosmospy/issues/24#issuecomment-1005589846) i.e. remove these functions and instead make `get_tx_bytes` public. Then document in the README how to broadcast via either the Cosmos REST API or Tendermint RPC API.

Reply:
This would also require some more changes as I think this would make the whole sync mode in typing obsolete and also the param which will be set on the transaction creation

Reply:
Yeah let's remove all of that.

Reply:
I am currently on it. Also to fix the tests then to use the tx_bytes function then

Reply:
Actually. You had a get_pushable method. I totally agree not to use requests or any broadcast method in it but I have to admit that the get_pushable method is handy to use

Reply:
It can be handy but not strictly necessary. Given that cosmospy is a "tool for _offline_ transaction signing" I think it was always a bit ugly for it to contain information about the broadcast sync modes. Now that we have more than one broadcast APIs I dislike this bloat even more.

I know things like this may be convenient to hobbyists but enterprise users want to user their own networking libraries and configurations. Hence it makes sense to have a low-level offline library. It may make sense to have another one that is as easy to use as possible, but I have no willingness to maintain that.

Reply:
Thank you very much for this explanation. I really appreciate it!